states/huntsman/twilight_flat_fielding.py

```diff
@@ -0,0 +1,76 @@
+"""
+State to handle the taking of calibration frames (evening and morning).
+"""
+
+
+def get_cameras(pocs):
+    '''
+    Get lists of narrow and broad band cameras.
+    '''
+    narrow_band_cameras = list()
+    broad_band_cameras = list()
+    for cam_name, cam in pocs.observatory.cameras.items():
+
+        # This is a hack. There should be a narrow property in the config...
+        if cam.filter_type.lower().startswith('ha'):
+            narrow_band_cameras.append(cam_name)
+        else:
+            broad_band_cameras.append(cam_name)
+
+    return narrow_band_cameras, broad_band_cameras
+
+
+def wait_for_twilight(pocs):
+    '''
+    Wait for twilight. Temporary solution until something better is found.
+
+    Twilight when Sun between flat and focus horizons.
+    '''
+    pocs.logger.debug('Waiting for twilight...')
+    while pocs.is_safe(horizon='flat'):
+        if pocs.is_dark(horizon='focus'):
```

Comment:
It still seems like if you start observing and get into this state after the sun has passed `focus` that this will then wait until the morning.

Reply:
Yes it will - and that's sort of desirable behaviour in the morning. At the moment the logic for entering the state should prevent it entering too early.